client/components/fields/editor/EventRelatedPlannings/EmbeddedCoverageForm.tsx

```diff
@@ -81,28 +81,43 @@ function getLanguagesForCoverage(
     };
 }
 
-export class EmbeddedCoverageFormComponent extends React.PureComponent<IProps> {
+interface IState {
+    userList: Array<IUser>;
+    selectedDeskId?: string;
+}
+
+export class EmbeddedCoverageFormComponent extends React.PureComponent<IProps, IState> {
     constructor(props) {
         super(props);
 
+        this.state = {
+            userList: this.props.users,
+            selectedDeskId: null,
+        };
+
         this.onDeskChange = this.onDeskChange.bind(this);
         this.onStatusChange = this.onStatusChange.bind(this);
         this.onUserChange = this.onUserChange.bind(this);
         this.onLanguageChange = this.onLanguageChange.bind(this);
     }
 
     onDeskChange(deskId?: IDesk['_id']) {
-        const newDesk = deskId == null ? null : this.props.desks.find((desk) => desk._id === deskId);
+        const newDesk = deskId == null || deskId == '' ? null : this.props.desks.find((desk) => desk._id === deskId);
 
         const updates: Partial<ICoverageDetails> = {
             desk: newDesk,
             filteredUsers: getUsersForDesk(newDesk, this.props.users),
         };
 
         if ((this.props.coverage.language ?? '').length < 1) {
-            updates.language = newDesk.desk_language;
+            updates.language = newDesk?.desk_language;
```

Comment:
is `updates.language = undefined` intentional? If not, add a condition one line above to make sure it's not null before adding it to updates.

Reply:
It's intentional

Reply:
could you add a fallback to `null`? Then it would be more obvious it's intentional

Reply:
If it doesn't exist at all what's the reason?

Reply:
The reason is to make it clear to the person reading the code that your intention is to remove a value vs that you forgot to handle the scenario where it's not defined